src/contract.rs

```diff
@@ -1043,9 +1034,8 @@ fn modify_contract(
         return Err(ContractError::Unauthorized);
     }
 
-    let ask_storage = get_ask_storage_read(deps.storage);
-    let ask_orders: Vec<AskOrderV1> = ask_storage
-        .range(None, None, Order::Ascending)
+    let ask_orders: Vec<AskOrderV1> = ASKS_V1
+        .range(deps.storage, None, None, Order::Ascending)
```

Comment:
Gas reduction is in https://github.com/FigureTechnologies/ats-smart-contract/pull/121/